### PR TITLE
Add a simple linux / powershell docker container to run the sdk

### PR DESCRIPTION
**Pull request by rickjuryxero**
This is configuration to create a simple docker container that will download and import the sumo powershell sdk inside a linux powershell container.

You can open it interactively with docker run -it or add your own scripts as part of your own docker based automation project. 

I have found this useful for automation of some of our collectors and source configuration via the sumo API.

### CODE REVIEW

**Comment by bin3377** [comment]
@rickjuryxero  Thanks for your contribution! 
Since this is a supporting tool, could you please move it to `/tools/` folder for organization and we can add a link in the main README.md point to it.

File: tools/docker.sumologic-powershell-sdk/profile.ps1 [hunk 1]
```diff
@@ -0,0 +1,36 @@
+Import-Module ./SumoLogic.psm1
+
+if ($env:SHOW_CONNECTION_INFO -eq $true) {
```
Comment (bin3377) [inline tools/docker.sumologic-powershell-sdk/profile.ps1]: it's better using either `$env` or `$Env` for consistency

File: tools/docker.sumologic-powershell-sdk/profile.ps1 [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    write-host "Available commands in sumologic loaded module are:"
+    get-command -Module Sumologic
+}
+
+if ($Env:SUMO_SESSION) {
+    if (!$Env:SUMO_DEPLOYMENT -or !$Env:SUMO_ACCESS_ID -or !$Env:SUMO_ACCESS_KEY  ) {
```
Comment (bin3377) [inline tools/docker.sumologic-powershell-sdk/profile.ps1]: Looks like no matter `true` or `false` in `$env:SUMO_SESSION`, all 3 vars are required? Maybe we should move this prompt to outside?

File: tools/docker.sumologic-powershell-sdk/profile.ps1 [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      
+        try { 
+            write-verbose "establishing connection to: $deployment"
+            $session = New-SumoSession -Deployment $Env:SUMO_DEPLOYMENT -AccessId $Env:SUMO_ACCESS_ID -AccessKey $Env:SUMO_ACCESS_KEY
+        }                   
+        catch { 
+            $ErrorMessage = $_.Exception.Message
+            $FailedItem = $_.Exception.ItemName
+            $hint = "An error occurred creating connection to SumoLogic API."
+            write-host "$hint`n$ErrorMessage`n$FailedItem"; exit 4
```
Comment (bin3377) [inline tools/docker.sumologic-powershell-sdk/profile.ps1]: better change to `write-error`

File: tools/docker.sumologic-powershell-sdk/profile.ps1 [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        }
+
+        if ($env:SHOW_CONNECTION_INFO -eq $true) {
+            write-host "Connected to $($session.endpoint)"
+        }
+    }
+    
+}
+else {
+    write-host 'WARNING! SUMO_SESSION is not set you must make your own with New-SumoSession '
```
Comment (bin3377) [inline tools/docker.sumologic-powershell-sdk/profile.ps1]: change to `write-warning`

File: tools/docker.sumologic-powershell-sdk/readme.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+```
+docker run --env SUMO_DEPLOYMENT=US2 --env SUMO_ACCESS_ID=$SUMO_ACCESS_ID --env SUMO_ACCESS_KEY=$SUMO_ACCESS_KEY --env SHOW_CONNECTION_INFO=false -entrypoint=pwsh  sumologic-powershell-sdk:latest -c get-collector -limit 1 -offset 1
+```
+
+# demo scripts
+The container puts two demo scripts in /home/
+- demo.ps1
+- TestNewConnection.ps1
+
+# to build run
```
Comment (bin3377) [inline tools/docker.sumologic-powershell-sdk/readme.md]: > How to Build Docker image

File: tools/docker.sumologic-powershell-sdk/readme.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
+# docker container to run sumologic powershell sdk on .net core /linux
```
Comment (bin3377) [inline tools/docker.sumologic-powershell-sdk/readme.md]: > Docker container to run Sumo Logic PowerShell SDK with .NET Core on Linux

Not a big problem but in document (here and later paragraph) it's better mention the brands/productions as it's well-known form
as
`PowerShell`, `.NET Core`, `Linux`, `Docker`, `Sumo Logic`, etc.

**Review by bin3377 — COMMENTED** [review]
Overall LGTM, with few minor comments

**Comment by rickjuryxero** [comment]
Those are all good points. I think I've addressed the issues you raised.

**Comment by bin3377** [comment]
Thanks for tweaking on them and quick response. Will merge it.

**Pull request merged**